### DISCUSSION TITLE
Module authoring

### DISCUSSION THREAD

**Discussion by raphmarot** [nuxt-modules/i18n#1793] [Q&A]
Hello,

I try to translate a Nuxt 3 module.

The extend-messages hook described here [https://v8.i18n.nuxtjs.org/guide/extend-messages](https://v8.i18n.nuxtjs.org/guide/extend-messages) is very useful, but i cannot get the playgound to work.

How to import properly i18n in the module's playgound?

Thanks!

**Comment by ineshbose** [comment]
Hi @Vincebe,

Thanks for opening a discussion. By getting the playground to work, do you mean like on StackBlitz? (https://stackblitz.com/github/nuxt-modules/i18n)

Typically the scripts StackBlitz runs is `pnpm install && pnpm dev` (which should work, but for a tagged release of `@nuxtjs/i18n`) while the official Nuxt module template suggests `npm i && npm run dev:prepare && npm run dev` to test local changes - I feel the playground hasn't been setup for this though (/cc @kazupon)

**Reply by raphmarot** [reply]
Hi @ineshbose 

Thanks.

No, i mean the official nuxt module template. Tried a basic i18n config, but can't find a way to import useI18n in app.vue.

**Reply by ineshbose** [reply]
Could you share your code please?

**Reply by Yves852** [reply]
I've made a repro using nuxt module.
https://github.com/Yves852/nuxt-i18n-with-module

As I understood, @nuxtjs/i18n (8.0.0-beta.7) is supposed to be installed on the main but not on the module. I've managed to use the local locale with playground while main project should use module locale. I'm able to overwrite from main but I fail to use module locale on main project. 

Also in module.ts `nuxt.hook('i18n:extend-messages', async (additionalMessages, localeCodes)` return a ts error `not assignable to parameter of type 'HookKeys<NuxtHooks>`. It makes the command `pnpm prepack` fail but adding a ts ignore allow the prepack to run. But module locales are not available.

Should I open a bug report?

NB: the repro show a directly included module but we aim to author hosted modules.
